File: Restaurant/app/src/main/java/com/example/restaurant/data/model/Table.kt
package com.example.restaurant.data.model

import com.google.gson.annotations.SerializedName

data class Table(
    @SerializedName("TableID")
    val id: Int,
    @SerializedName("TableNumber")
    val tableNumber: String,
    @SerializedName("Status")
    val status: String,
    @SerializedName("Seats")
    val seats: Int,
    @SerializedName("Location")
    val location: String,
    @SerializedName("created_at")
    val createdAt: String?,
    @SerializedName("updated_at")
    val updatedAt: String?
)

File: hari-restaurant/app/Models/Table.php
<?php

namespace App\Models;

use App\Models\Reservation;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;


class Table extends Model
{
    use HasFactory;
    protected $table = 'table';
    protected $primaryKey = 'TableID';

    protected $fillable = [
        'TableNumber',
        'Seats',
        'Location',
        'Status'
    ];

    /**
     * Quan hệ với Reservation.
     */
    public function reservations()
    {
        return $this->hasMany(Reservation::class, 'TableID');
    }
    public function activeReservations()
    {
        return $this->hasMany(Reservation::class, 'TableID')
            ->where('Status', 'Đã xác nhận');
    }
}
